### PR TITLE
Run SAML adapter tests with EAP 8

### LINKED ISSUE #24168: Run SAML  adapter tests with EAP 8
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

testsuite

### Describe the bug

When running SAML adapter tests for EAP8 an error is thrown. It might be related to the recent migration from javax to jakarta.

### Version

latest

### Expected behavior

Tests are executed successfully

### Actual behavior

The following error occurs when deploying the test app:

`15:56:57,666 WARN  [org.jboss.modules.define] (MSC service thread 1-1) Failed to define class org.keycloak.adapters.saml.servlet.SamlFilter in Module "deployment.employee2.war" from Service Module Loader: java.lang.NoClassDefFoundError: Failed to link org/keycloak/adapters/saml/servlet/SamlFilter (Module "deployment.employee2.war" from Service Module Loader): javax/servlet/Filter`

### How to Reproduce?

- Build Keycloak
- `mvn -f testsuite/integration-arquillian/servers/pom.xml clean install -DskipTests -s settings-prod.xml -Dinsecure.repositories=WARN -Papp-server-eap`
- `mvn -f testsuite/integration-arquillian/tests/base/pom.xml clean install -s settings-prod.xml -Dinsecure.repositories=WARN -Dbrowser=chrome -Dwebdriver.chrome.driver=/usr/bin/chromedriver -Dtest=SAMLFilterLoginResponseHandlingTest#testAttributes -Pauth-server-quarkus -Papp-server-eap`

### Anything else?

WIP branch: https://github.com/miquelsi/keycloak/tree/main_eap8

### PR DESCRIPTION
**Pull request by mabartos**
By providing these changes, the SAML Adapter tests execution should be working with EAP8 with the usage of Keycloak SAML Adapter Galleon Feature Pack. 

It'd be great to include it in the pipeline. 

@miquelsi @pskopek  Could you please check it?

### CODE REVIEW

File: pom.xml
Comment (mabartos) [inline pom.xml:225]: Not productized yet. Need to verify the impact to the approach.

**Comment by mabartos** [comment]
One issue with Servlet Filter tests.

**Comment by mabartos** [comment]
This PR is ready to review. 

EAP8 works as expected; only one additional test fails in the pipeline, but could be solved in a follow-up task. The other tests fail even for EAP7, and should be tracked (cc: @miquelsi ).

Pipeline run: https://keycloak-pipeline.com/view/Keycloak/job/universal-test-pipeline-adapters/1955/testReport/

Follow-up tasks:

- [ ] https://github.com/keycloak/keycloak/issues/24876
- [ ] https://github.com/keycloak/keycloak/issues/24878

The @keycloak/testing team might solve follow-up tasks as it's more related to the testsuite. 

@miquelsi Could you please check it? Thanks

**Comment by mabartos** [comment]
It'd also be beneficial to have more eyes on this, specifically from the @keycloak/core team. Could somebody from the @keycloak/core take a look at it? Thanks :))

**Review by miquelsi — APPROVED** [review]
Great job, thanks @mabartos!

**Comment by mposolda** [comment]
@mabartos @miquelsi Thanks for the PR and for pointing this!

I've merged the related PR from WildFly team https://github.com/keycloak/keycloak/pull/24438. @mabartos Is it possible to rebase your PR on top of the latest main to make sure that tests are still passing fine with that PR 24438 together with your changes? (Not 100% sure if there is something in that PR, which can break yours, but just to doublecheck...)

@pskopek Is this PR ok to you?

**Comment by mabartos** [comment]
@mposolda @miquelsi @pskopek Tests execution completed as expected even after the rebase[1].

The only remaining task is to get @pskopek's approval.

[1] https://keycloak-jenkins.com/view/Keycloak/job/universal-test-pipeline-adapters/1959/testReport/

**Review by pskopek — APPROVED** [review]
Looks good

**Review by mposolda — APPROVED** [review]

**Pull request merged**
